crates/worker/src/executor/parameter_server.rs

```diff
@@ -164,96 +193,30 @@ impl JobExecutor for ParameterServerExecutor {
-                                                    result_tensor_file_name.as_path(),
-                                                ) {
-                                                    Ok(_) => {
-                                                        tracing::debug!("Updated averaged tensors");
-                                                    }
-                                                    Err(e) => {
-                                                        tracing::warn!(error = ?e, "Failed to serialize averaged tensors; keeping previous aggregation");
-                                                    }
-                                                }
-                                            } else {
-                                                tracing::warn!("Failed to mmap averaged tensors; keeping previous aggregation");
-                                            }
-
-                                            if let Err(e) = std::fs::remove_dir_all(work_dir.join("avg").join(name.as_str())) {
-                                                tracing::warn!(error = ?e, "Failed to cleanup avg directory");
-                                            }
-
-                                            result_tensor_file_name.clone()
-                                        }
-                                    }
-                                    Err(e) => {
-                                        tracing::warn!(error = ?e, "Failed to mmap new tensor; keeping previous aggregation");
-
-                                        // Keep previous aggregation file
-                                        result_tensor_file_name.clone()
-                                    }
+                                // Average the new tensor with an existing one
+                                let average_op = |a: &Tensor, b: &Tensor|{
+                                    // Compute (a + b) / 2.
+                                    (a + b).and_then(|t| t / 2.)
+                                };
+                                if let Err(e) = apply_tensor_op(
+                                    &file_name,
+                                    &result_tensor_file_name,
+                                    &temporary_tensor_file_name,
+                                    &tmp_dir_name,
+                                    &device,
+                                    average_op,
+                                )
+                                .await{
+                                    tracing::warn!(error = ?e, "Failed to average results");
                                 }
-                            }}).await.expect("averaging tasks runs to completion")
+                                fs::copy(temporary_tensor_file_name.clone(), result_tensor_file_name.clone()).await.expect("file can be copied");
+                                fs::remove_file(temporary_tensor_file_name).await.expect("Delete temporary average");
```

Comment:
Why can't we just rename (mv) the file once we're done? Do we need to copy/duplicate them?

```suggestion
fs::rename(&temporary_tensor_file_name, &result_tensor_file_name)?;
```